packages/react/src/components/MenuButton/MenuButton.tsx

```diff
@@ -0,0 +1,305 @@
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+import * as _ from 'lodash'
+import * as customPropTypes from '@stardust-ui/react-proptypes'
+
+import {
+  AutoControlledComponent,
+  RenderResultConfig,
+  applyAccessibilityKeyHandlers,
+  getOrGenerateIdFromShorthand,
+  commonPropTypes,
+  StyledComponentProps,
+} from '../../lib'
+import { ShorthandValue, ComponentEventHandler } from '../../types'
+
+import { Accessibility } from '../../lib/accessibility/types'
+import { createShorthandFactory } from '../../lib/factories'
+import Popup, { PopupProps, PopupEvents, PopupEventsArray } from '../Popup/Popup'
+import { Menu, MenuItemProps, MenuProps, Ref } from '../..'
+import { contextMenuBehavior } from '../../lib/accessibility'
+import { focusMenuItem, focusNearest } from './focusUtils'
+import { ALIGNMENTS, POSITIONS, PositioningProps } from '../../lib/positioner'
+
+export interface ContextMenuSlotClassNames {
+  menu: string
+}
+
+export interface ContextMenuProps extends StyledComponentProps<ContextMenuProps>, PositioningProps {
+  /**
+   * Accessibility behavior if overridden by the user.
+   * @default contextMenuBehavior
+   */
+  accessibility?: Accessibility
+
+  /** Additional CSS class name(s) to apply.  */
+  className?: string
+
+  /** Initial value for 'open'. */
+  defaultOpen?: boolean
+
+  /** Whether the Popup should be rendered inline with the trigger or in the body. */
+  inline?: boolean
+
+  /** Existing document the popup should add listeners. */
+  mountDocument?: Document
+
+  /** Existing element the popup should be bound to. */
+  mountNode?: HTMLElement
+
+  /** Delay in ms for the mouse leave event, before the popup will be closed. */
+  mouseLeaveDelay?: number
+
+  /** Events triggering the popup. */
+  on?: PopupEvents | PopupEventsArray
+
+  /** Defines whether popup is displayed. */
+  open?: boolean
+
+  /**
+   * Event for request to change 'open' value.
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onOpenChange?: ComponentEventHandler<PopupProps>
+
+  /** A popup can show a pointer to trigger. */
+  pointing?: boolean
+
+  /**
+   * DOM element that should be used as popup's target - instead of 'trigger' element that is used by default.
+   */
+  target?: HTMLElement
+
+  /** Element to be rendered in-place where the popup is defined. */
+  trigger?: JSX.Element
+
+  /** Whether the trigger should be tabbable */
+  shouldTriggerBeTabbable?: boolean
+
+  /** Ref for Popup content DOM node. */
+  contentRef?: React.Ref<HTMLElement>
+
+  /** Shorthand for menu configuration */
+  menu?: ShorthandValue<MenuProps>
+
+  /** Determines if the MenuButton behaves as context menu */
+  contextMenu?: boolean
```

Comment:
Why it's required? Why we can't use `const isContextMenu = this.props.on === 'content'`?

Reply:
no, that is a different use case. contextMenu is for any element where you can click on. on=context is just a regular MenuButton which opens on context. The use case: For reactions, we want the popup appear on hover and on contextmenu so that both screen reader users can activate it (screen readers do not have a concept of hover). On click, an action of selecting/unselecting the reaction should be executed. So it will be something like:
```
<MenuButton
  trigger={reaction}
  menu={...}
  on=['context', 'hover']
  />
```

Whereas contextMenu will be something like 
```
<MenuButton
  trigger={<TreeItem ... />}
  menu={...}
  on=['context']
  />
```

It is currently not documented at all, but I will add the explanation to the examples